lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2847,8 +2848,12 @@ public function createEntity($className, array $data, &$hints = [])
 
                     $this->originalEntityData[$oid][$field] = $newValue;
                     $class->reflFields[$field]->setValue($entity, $newValue);
+                    if ( ! $assoc['inversedBy'] && ! $assoc['type'] && ! isset($targetClass->associationMappings[$assoc['inversedBy']])) {
+                        throw new ORMException("The column " . $assoc['inversedBy'] . " is needed by " . $targetClass->getName() . " entity and should be are on the database", 1);
```

Comment:
There should be unit tests for this piece of code, but did you at least test it manually? (Beware, it's a trick question)

Reply:
yeah i test it manually it work with the good error message but travis test didn't expect a throw error, then i add the two negation for target more precisely the error. Then i think the code error should be a great idea

Reply:
Also, I think this should be moved inside the if below, and you should only check for `isset($targetClass->associationMappings[$assoc['inversedBy']])`

Reply:
are u french ? (in this case will be more practical to speak in french cae me too)

Reply:
Sorry, we can't, this needs to be understandable for everyone. If you're going to contribute to open source, using english is a must, so let's start as soon as possible, shall we?

>  did i something wrong ?

Well `ORMException` is an interface, for starters, so you can't throw it. Also, checking `! $assoc['type']` has no meaning, while the original expression `$assoc['type'] & ClassMetadata::ONE_TO_ONE` you copied it from has some. I suggest you read about the `&` operator to understand what it does. https://www.php.net/manual/en/language.operators.bitwise.php

Reply:
Ok i forgot binary operator ( i don't really use it except in a game ), for the throw, i see how  to do it like add a method in the ORMExeption. to throw the exeption with a static method like line 897 UnitOfWork.php